### ISSUE TITLE
Cython 0.28 Python 3.7 Traceback Regression

### ISSUE THREAD

**Issue by MSeal** [cython/cython#2477]
With any Cython 0.28.x I get an error when trying to install the generated cpp from Python 3.7 (but not 3.6). This issue was fixed for Python 3.7 late in 0.27.x versions but seems to have been reintroduced:

```
# Activate python 3.6
. ~/py36venv/bin/activate
pip install -U cython==0.28.3
git clone https://github.com/MSeal/cython_hunspell
cd cython_hunspell
python setup.py build_ext
# Activate python 3.7
. ~/py37env/bin/activate
python setup.py install
```
And see the error below:
```
hunspell/hunspell.cpp:9504:22: error: ‘PyThreadState {aka struct _ts}’ has no member named ‘exc_state’; did you mean ‘exc_type’?
     tmp_tb = tstate->exc_state.exc_traceback;
```
Changing to build to cython==0.27 results in a successful install

**Comment by MSeal** [comment]
This appears to be a difference between A2 and A3. 3.7a2 does not expect exc_state while 3.7a3 does. See https://github.com/python/cpython/compare/v3.7.0a2...v3.7.0a3 . Ubuntu 17.10 is installing 3.7a2 from it's default repositories so I hit this issue.

**Comment by scoder** [comment]
I wouldn't really call it a regression, given that we're dealing with alpha releases of CPython here. But thanks for your investigations. Here's a fix: 9e74fb6c3910200c24cbfc56c59f5f6eb1ad5e57

**Comment by MSeal** [comment]
Yea I originally had it compiling with 0.27.3 so thought it had been working before, but that wasn't handling exceptions in the code path at all once I tested it more today, so it really wasn't a regression in any way. Thanks for the quick fix.
